src/allotropy/testing/utils.py

```diff
@@ -196,3 +190,9 @@ def validate_contents(
                 msg = f"Mismatch between actual and expected for '{expected_file}', writing expected output because 'write_actual_to_expected_on_fail=True'\n\n{e}"
                 raise AssertionError(msg) from e
         raise
+
+    # Ensure that allotrope_dict can be structured back into a python model.
```

Comment:
Note: this is so if a test fails with different content than on local, we see the diff in content before other failures.